DirectProgramming/DPC++FPGA/Tutorials/Features/dsp_control/README.md

```diff
@@ -0,0 +1,186 @@
+| `ac_fixed`             | Constant mul
+
+**NOTE:** _constant mul_ means one operand of the multiplication is a constant.
+
+### Global Control
+The `-Xsdsp-mode` command-line option sets implementation preference of math operations that support DSP control in all kernels. It has three valid options:
+| Option                           | Explanation
+---                                |---
+| `-Xsdsp-mode=default`            | This is the default. The compiler determines the implementation based on datatype and math operation.
+| `-Xsdsp-mode=prefer-dsp`         | Prefer math operations to be implemented in **DSPs**. If a math operation is implemented by DSPs by default, you will see no difference in resource utilization or area. Otherwise, you will notice decrease in the usage of soft-logic resources and increase in the usage of DSPs.
+| `-Xsdsp-mode=prefer-softlogic`   | Prefer math operations to be implemented in **soft-logic**. If a math operation is implemented without DSPs by default, you will see no difference in resource utilization or area. Otherwise, you will notice decrease in the usage of DSPs and increase in the suage of soft-logic resources.
+
+### Local Control
+The library function `math_dsp_control<Preference::<enum>, Propagate::<bool>>([&]{})` provides block scope local control in one single kernel. A reference-capturing lambda expression is passed as the argument to this library function. Inside the lambda expression, implementation preference of math operations that support DSP control will be determined by two template arguments.
+
+The first template argument `Preference` is an enum with three valid options:
+| Option                           | Explanation
+---                                |---
+| `Preference::DSP`                | This is the default. Prefer math operations to be implemented in **DSPs**. Its behavior on a math operation is equivalent to global control `-Xsdsp-mode=prefer-dsp`.
+| `Preference::Softlogic`          | Prefer math operations to be implemented in **soft-logic**. Its behavior on a math operation is equivalent to global control `-Xsdsp-mode=prefer-softlogic`.
+| `Preference::Compiler_default`   | Compiler determines the implementation based on datatype and math operation. Its behavir on a math operation is equivalent to global control `-Xsdsp-mode=default`.
+
+The second template argument `Propagate` is a boolean that determines propagation of the first argument `Preference` to functions called in the lambda expression:
+| Option  | Explanation
+---       |---
+| `On`    | This is the default. `Preference` is propagated to all functions called in the lambda expression, and affects all math operations on the call graph in the lambda expression until a nested `math_dsp_control<>()` call stops the propatation.
+| `Off`   | `Preference` is not propagated to functions called in the lambda expression. So `Preference` only applies to math operations directly used in the lambda expression.
+
```

Comment:
You may also want to show that you can call the function with no template params and you'll get the DSP option with propagate. A couple of code snippets could be useful to have in the README as well.

Reply:
Added a code snippet.